File: services/dotnet-microservices/PaymentService/Domain/Models/Payment.cs
namespace PaymentService.Domain.Models;

public record Payment(
    Guid Id,
    Guid OrderId,
    Guid UserId,
    decimal Amount,
    string Currency,
    string Status,
    string? PaymentMethod,
    string? TransactionId,
    string? FailureReason,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

public record Refund(
    Guid Id,
    Guid PaymentId,
    decimal Amount,
    string? Reason,
    string Status,
    DateTime CreatedAt
);

public record ProcessPaymentRequest(
    Guid OrderId,
    Guid UserId,
    decimal Amount,
    string? PaymentMethod,
    string Currency = "USD"
);

public record RefundRequest(Guid PaymentId, decimal Amount, string? Reason);

public static class PaymentStatuses
{
    public const string Pending = "pending";
    public const string Processing = "processing";
    public const string Completed = "completed";
    public const string Failed = "failed";
    public const string Refunded = "refunded";
}

File: services/nodejs-microservices/payment-service/src/domain/models/Payment.ts
export interface Payment {
  id: string;
  orderId: string;
  userId: string;
  amount: number;
  currency: string;
  method: PaymentMethod;
  status: PaymentStatus;
  providerTransactionId?: string;
  failureReason?: string;
  createdAt: Date;
  updatedAt: Date;
}

export enum PaymentMethod {
  CREDIT_CARD = 'CREDIT_CARD',
  DEBIT_CARD = 'DEBIT_CARD',
  PAYPAL = 'PAYPAL',
  BANK_TRANSFER = 'BANK_TRANSFER',
}

export enum PaymentStatus {
  PENDING = 'PENDING',
  PROCESSING = 'PROCESSING',
  COMPLETED = 'COMPLETED',
  FAILED = 'FAILED',
  REFUNDED = 'REFUNDED',
}

export interface Refund {
  id: string;
  paymentId: string;
  amount: number;
  reason?: string;
  status: RefundStatus;
  createdAt: Date;
}

export enum RefundStatus {
  PENDING = 'PENDING',
  COMPLETED = 'COMPLETED',
  FAILED = 'FAILED',
}
